### PR TITLE
New Components: Projects, Project, and Review.

### PR DESCRIPTION
**Pull request by McCambley**
Hey all!
A lot going on here so I'll do my best to break it down. 

Three new Components: 
1. `Projects` - The component that will be featured on the `main` page as well as on each `profession` page
2. `Project` - A component that is rendered into the projects component from server data
3. `Review` - A component that is rendered from an array attached to the data fetched from the above server request

There's also some other changes to note:
1. Adjusted from CSS in `SellingPoints` in order to remove horizontal scrolling
2. Added a title object in `/arrays` that will provide data to each instance of the Projects component in order to provide the correct title
3. Made some adjustments to `delegate-tasks.js` in order to mesh a tad more smoothly with the intended functionality of each project card

Future steps here:
1. Render components with actual server data
2. Test edge cases (super long reviews, weird sized images from the server, etc.
3.  Work with colin to connect the `Projects` component to the `profession` pages

I'm probably missing something else but that's all I've got for now!

[Here's a demo of the behavior with hopefully minimal rambling](https://www.loom.com/share/d433a32b13864e829414347f0afae6e0) 🎉

### CODE REVIEW

File: src/arrays/delegate-tasks.js [hunk 1]
```diff
@@ -1,177 +1,615 @@
...
-        '“The Data Analytics students have highlighted a few languages which show a growing demand for translation and expanded our list of keywords to detect more translation-related user reviews,” Demid said. “The Data Science students tested a wide range of natural language processing algorithms, including deep machine learning..."',
-      link: 'Read full review',
-    },
-    'bottom-left': {
-      avatars: [avatar, avatar, avatar],
-      title: 'Students team',
-      subtitle: 'Maria Wright, Maria Wright, Enyel Sequeira',
-      description:
-        '“I learned so much from it,” Xia said. “Not just about new packages and methods of analysis and reporting, but also workflow management, how to clarify task requirements, collaborate with others, and more. Those are the kind of experiences I can put on my resume, and that’s awesome too.”',
+      link: 'https://www.example.com/',
```
Comment (thinklikeadesigner) [inline src/arrays/delegate-tasks.js:20]: Totally forgot that links also need a link. Thank you for adding this!

File: src/arrays/delegate-tasks.js [hunk 2]
```diff
@@ -1,177 +1,615 @@
...
+        avatars: [{ id: 1, src: avatar }],
+        title: 'Maxim Antonenko',
+        subtitle: 'Ant Robotics CEO',
+        description:
+          ' “The Data Analytics students have highlighted a few languages which show a growing demand for translation and expanded our list of keywords to detect more translation-related user reviews,” Demid said. “The Data Science students tested a wide range of natural language processing algorithms, including deep machine learning algorithms, to build a robust model that accurately predicts if a review praises, criticizes or requests game translation.”',
+      },
+      {
+        id: 12,
+        avatars: [
+          { id: 1, src: avatar },
```
Comment (thinklikeadesigner) [inline src/arrays/delegate-tasks.js:35]: This is great. It's a list of images, so they definitely needed id's so you can map over it. Thanks for catching that

File: src/arrays/why-choose-us.js [hunk 3]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable import/prefer-default-export */
```
Comment (thinklikeadesigner) [inline src/arrays/why-choose-us.js:1]: there's an export on line 5 and line 32

You can do it either way, but the one at the bottom complies with eslint's default export rule

Most important thing is to remember to either add or take away {} on  import based on the one you choose

File: src/components/footer/Footer.js [hunk 4]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable jsx-a11y/anchor-is-valid */
 import React from 'react';
 import './Footer.css';
```
Comment (thinklikeadesigner) [inline src/components/footer/Footer.js:4]: Great use of rel="noopener noreferrer"!!

File: src/components/projects/Projects.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+  function updateCourse(course) {
+    setCourse(course);
+  }
+
+  function showMore() {
+    if (!isExpanded) {
+      setProjects(focusedCourse);
+    } else {
+      setProjects(focusedCourse.slice(0, 2));
```
Comment (thinklikeadesigner) [inline src/components/projects/Projects.js:34]: Great job on the show more button!

Since there is an unknown amount of projects in the cms, you can use a count variable for the second argument in your slice method

I did something similar here https://github.com/thinklikeadesigner/news-explorer-frontend/blob/main/src/components/CardsList/CardsList.js

File: src/components/review/Review.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+import React from 'react';
+import './Review.css';
+
+export default function Review({ data }) {
+  const { avatars, description, subtitle, title } = data;
+
+  const [displayedReview, setReview] = React.useState(description);
+  const [isReviewExpanded, setExpanded] = React.useState(false);
+
+  const lengthLimit = 290;
```
Comment (thinklikeadesigner) [inline src/components/review/Review.js:11]: nice avoidance of magic numbers here 😎

File: src/components/review/Review.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+  const shortenedReview = `${description.slice(0, lengthLimit)}..."`;
```
Comment (thinklikeadesigner) [inline src/components/review/Review.js:12]: This is genius! I implemented this with css in a different project with something like a multiline overflow ellipsis, but I didn't even consider limiting the string itself 🤯

**Review by thinklikeadesigner — CHANGES_REQUESTED** [review]
There's really only two changes, the double use of the word export in project-page-titles.js and why-choose-us.js and making show more projects button dynamic for n number of projects.


The rest of the comments is just stuff I thought was clever :)

File: src/components/projects/Projects.js (hunk 5 above)
Comment (cjmaret) [reply src/components/projects/Projects.js:34]: @thinklikeadesigner , very smart! this already looked amazing, but that's a good idea

File: src/arrays/delegate-tasks.js (hunk 1 above)
Comment (McCambley) [reply src/arrays/delegate-tasks.js:20]: Of course! I omitted linkText for the longest time until I realized that the the link-text depends on the curriculum, so thank _you_ for adding that portion. Seriously, I cannot overstate how nice it was to have this data all set and ready to go in such a beautiful manner, you rock.

File: src/arrays/why-choose-us.js (hunk 3 above)
Comment (McCambley) [reply src/arrays/why-choose-us.js:1]: Ohh good catch, I'll remove the redundant ones.

File: src/components/footer/Footer.js (hunk 4 above)
Comment (McCambley) [reply src/components/footer/Footer.js:4]: Thank thank thank! Big ups to chromes lighthouse feature for teaching my that trick.

**Comment by McCambley** [comment]
Updated the projects list to hide the `Show more` button when the array is small. I've commented out two of the four projects in the `webDevelopment` projects array so that you can see how the functionality works. The button shouldn't display on the webDev projects section, but should for the other two courses.

File: src/components/footer/Footer.js (hunk 4 above)
Comment (thinklikeadesigner) [reply src/components/footer/Footer.js:4]: I didn't know you could run lighthouse directly from dev tools, that's awesome!

File: src/components/review/Review.js (hunk 7 above)
Comment (thinklikeadesigner) [reply src/components/review/Review.js:12]: 💯

**Review by thinklikeadesigner — APPROVED** [review]
